### PR TITLE
chore: clean up type test setup

### PR DESCRIPTION
**Pull request by mrazauskas**
@eegli  Great to see TSTyche in use!

This PR removes unused dependencies and adds missing `tsconfig.json`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,7 +57,6 @@
     "@babel/preset-typescript": "^7.23.3",
     "@changesets/cli": "^2.27.1",
     "@preconstruct/cli": "^2.8.2",
-    "@tsd/typescript": "^5.3.3",
```
Comment (mrazauskas) [inline package.json:60]: TSTyche is using the same `typescript` you have installed in the repo.

File: test/types/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "types": []
+  },
+  "include": ["**/*"],
+  "exclude": []
```
Comment (mrazauskas) [inline test/types/tsconfig.json:7]: The root `tsconfig.json` does not included type test files. TSTyche is using compiler options from the nearest TSConfig which includes file in question. This is the same how code editors works.

If such TSConfig is not found, default options are set.

File: tstyche.config.json
Comment (mrazauskas) [inline tstyche.config.json:1]: All `*.tst.ts` files will be picked up. Perhaps it makes sense to have less files in the root directory?

**Comment by mrazauskas** [comment]
By the way, so nice to see that the `.toMatch()` matcher is useful. I was honestly in doubt about it. Now it is clear that it was good idea to have it around.

**Comment by eegli** [comment]
LGTM, thanks!

**Pull request merged**